pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go

```diff
@@ -13,6 +13,7 @@ import (
 )
 
 // PostgresClusterSpec defines the desired state of PostgresCluster
+// +kubebuilder:validation:XValidation:rule="(self.environment == 'production') ? self.backups.pgbackrest != null : true", message="Backups must be enabled for production PostgresCluster's."
```

Comment:
I went with the following `message`:
> Backups must be enabled in a production environment.

I also switch the `null` check to `has()`.  However, I did go one layer deeper with `has(self.backups.pgbackrest)` to protect against an empty `backups` section (`backups: {}`).  I don't love the error messages displayed when this does fail, since the validation rule itself essentially fails due to the missing field.  But for now this does appear to protect against everything we need it to, and it does seem like there is a fine line with keeping these errors pretty, and keeping the rules from getting too complex.

Reply:
I did attempt a bump to controller-gen `0.16.3`, but saw something a bit strange with our RBAC (I'll capture in a story if no thoughts in the Slack thread I started).  This did give me `fieldPath` and `reason`, though I found it was really only helpful if the user actually included the section (and the `reason`, while returned from the API, is displayed by `kubectl`).  But this could again be tied to further expanding the rule a bit (e.g. per my last comment) to get some better validation error messages/behavior.

Reply:
I went ahead and made the bump to `0.16.3`.  `fieldRef` and `reason` have now been added to the CEL validation marker as a result.

Additionally, I made one final tweak to the validation rule:

```
(self.environment == 'production') ? (has(self.backups) && has(self.backups.pgbackrest)) : true
```

Now I get a clean/consistent error whether `backups` is empty or missing:

**Missing `backups`**:
```bash
$ kubectl apply -n postgres-operator -f - <<EOF
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: hippo
spec:
  environment: production                                                            
  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:ubi8-16.3-1
  postgresVersion: 16
  instances:
    - name: instance1
      dataVolumeClaimSpec:
        accessModes:
        - "ReadWriteOnce"
        resources:
          requests:
            storage: 1Gi
EOF          
The PostgresCluster "hippo" is invalid: spec.backups: Required value: Backups must be enabled in a production environment.
```

**Empty `backups`:**
```bash
$ kubectl apply -n postgres-operator -f - <<EOF
apiVersion: postgres-operator.crunchydata.com/v1beta1
kind: PostgresCluster
metadata:
  name: hippo
spec:
  environment: production                                                            
  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:ubi8-16.3-1
  postgresVersion: 16
  instances:
    - name: instance1
      dataVolumeClaimSpec:
        accessModes:
        - "ReadWriteOnce"
        resources:
          requests:
            storage: 1Gi
  backups: {}
EOF
The PostgresCluster "hippo" is invalid: spec.backups: Required value: Backups must be enabled in a production environment.
```